app/src/main/java/org/schabi/newpipe/local/bookmark/BookmarkFragment.java

```diff
@@ -271,6 +303,39 @@ private void showDeleteDialog(final String name, final Single<Integer> deleteRea
                 .show();
     }
 
+    private void showLocalRenameDialog(PlaylistMetadataEntry selectedItem) {
+        final View dialogView = View.inflate(getContext(), R.layout.dialog_playlist_name, null);
+        EditText nameEdit = dialogView.findViewById(R.id.playlist_name);
+        nameEdit.setText(selectedItem.name);
+        nameEdit.setSelection(nameEdit.getText().length());
+
+        final AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(
+            getContext())
+            .setTitle(R.string.rename_playlist)
+            .setView(dialogView)
+            .setCancelable(true)
+            .setNegativeButton(R.string.cancel, null)
+            .setPositiveButton(R.string.rename, (dialogInterface, i) -> {
+                changeLocalPlaylistName(selectedItem.uid, nameEdit.getText().toString());
+            });
+        dialogBuilder.show();
+    }
+
+    private void changeLocalPlaylistName(long id, String name) {
+        if (localPlaylistManager == null) {
+            return;
+        }
+
+        Log.d(TAG, "Updating playlist id=[" + id +
+            "] with new name=[" + name + "] items");
```

Comment:
Can do.

Should all log statements use if(DEBUG)?
Since there are 3 occurences in LocalPlaylistFragment which are without.
[1](https://github.com/TeamNewPipe/NewPipe/blob/399e2626fb4e93abe93bc3a40e44af519f83ef88/app/src/main/java/org/schabi/newpipe/local/playlist/LocalPlaylistFragment.java#L391)  [2](https://github.com/TeamNewPipe/NewPipe/blob/399e2626fb4e93abe93bc3a40e44af519f83ef88/app/src/main/java/org/schabi/newpipe/local/playlist/LocalPlaylistFragment.java#L407)  [3](https://github.com/TeamNewPipe/NewPipe/blob/399e2626fb4e93abe93bc3a40e44af519f83ef88/app/src/main/java/org/schabi/newpipe/local/playlist/LocalPlaylistFragment.java#L475)

Reply:
Yes, please.

Reply:
done